theano/gof/op.py

```diff
@@ -779,10 +780,33 @@ class Op(utils.object2, PureOp, CLinkerOp):
     Convenience class to bundle `PureOp` and `CLinkerOp`.
 
     """
+    instances = {}
+
     def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
-        obj = object.__new__(cls)
+        key = list()
+        if args:
+            [key.append(i) for i in args]
+        if kwargs:
+            [key.append(v) for k,v in kwargs.iteritems()]
+
+        # not the best solution, a hack for now
+        for k in key:
+            if isinstance(k, collections.Hashable):
+               key.remove(k) 
+
+        # Since list is not hashable
+        key = tuple(key)
+        if key not in Op.instances:
```

Comment:
Yeah, i remember reporting this previously when i was trying to hash GpuElemwise. You recollect ? And I think that param could be anything (Any Op from `theano.scalar.basic`) and not only a dict as i reported above/